### PR TITLE
Display Fee rate column in mempool tx/tickets tables

### LINKED ISSUE #1251: mempool: show fee rate
Add a column to show fee rate for tickets and regular tx.

Sometimes when ticket purchasing is very active it helps to see ticket transactions and their fee rate (I assume fee rate and not just fee is what influences the priority of inclusion in blocks).

This can be a useful indicator for regular tx too.

### PR DESCRIPTION
**Pull request by peterzen**
Fixes #1251

### CODE REVIEW

File: views/mempool.tmpl [hunk 1]
```diff
@@ -183,12 +184,13 @@
                                     {{template "decimalParts" (float64AsDecimalParts .TotalOut 8 false)}}
                                   </td>
                                   <td class="mono fs15 text-right">{{.Size}} B</td>
+                                  <td class="mono fs15 text-right">{{.FeeRate}}/kB</td>
```
Comment (chappjc) [inline views/mempool.tmpl]: This gives `[ERR] EXPR: Template execute failure: template: mempool.tmpl:187:69: executing "mempool" at <.FeeRate>: can't evaluate field FeeRate in type types.MempoolTx`

Comment (peterzen) [reply views/mempool.tmpl]: Fixed in the PR.

**Comment by peterzen** [comment]
As the `MempoolTx` struct was changed in this PR the linter fails ATM - is there a `git` tag needed on `github.com/decred/dcrdata/explorer/types/v2` for the test suite to properly pick it up?

**Comment by chappjc** [comment]
Unfortunately due to the way we have CI running tests (first `cd`ing` into the module paths), each go.mod that requires explorer/types/v2 will need a replace.  Looks like that includes `mempool`, `pubsub`, `pubsub/types`, and `pubsub/democlient`.

File: mempool/collector.go [hunk 2]
```diff
@@ -127,9 +127,12 @@ func (t *MempoolDataCollector) mempoolTxns() ([]exptypes.MempoolTx, txhelpers.Me
 		//fee, _ := txhelpers.TxFeeRate(msgTx)
 		// log.Tracef("tx fee: GRM result = %f, msgTx = %f", tx.Fee, fee.ToCoin())
 
+		_, feeRate := txhelpers.TxFeeRate(msgTx)
+
 		txs = append(txs, exptypes.MempoolTx{
 			TxID:      msgTx.TxHash().String(),
 			Fees:      tx.Fee,
+			FeeRate:   feeRate.ToCoin(),
```
Comment (chappjc) [inline mempool/collector.go]: Also need something similar in mempool/monitor, (`(*MempoolMonitor).TxHandler`).

Comment (chappjc) [reply mempool/collector.go]: Looks like `(pgb *ChainDB).GetMempool` needs it too, but there seems to be a number of fields unset in that function too.  I don't recall a reason or see docs explaining why, so perhaps that's a bug.

File: mempool/collector.go [hunk 3]
```diff
@@ -127,9 +127,12 @@ func (t *MempoolDataCollector) mempoolTxns() ([]exptypes.MempoolTx, txhelpers.Me
 		//fee, _ := txhelpers.TxFeeRate(msgTx)
 		// log.Tracef("tx fee: GRM result = %f, msgTx = %f", tx.Fee, fee.ToCoin())
 
+		_, feeRate := txhelpers.TxFeeRate(msgTx)
```
Comment (chappjc) [inline mempool/collector.go]: Regarding the comment a few lines up, we may need special  handling for coinbase transactions.  I haven't checked.

File: views/mempool.tmpl [hunk 4]
```diff
@@ -168,6 +168,7 @@
                               <th>Transaction ID</th>
                               <th class="text-right">Total DCR</th>
                               <th class="text-right">Size</th>
+                              <th class="text-right">Fee rate</th>
```
Comment (chappjc) [inline views/mempool.tmpl]: For consistency with the other columns, best to use title case (i.e. Fee Rate).

File: mempool/collector.go (hunk 3 above)
Comment (peterzen) [reply mempool/collector.go]: Can there be a case where a coinbase TX gets added to the mempool?

**Comment by chappjc** [comment]
BTW, dcrdata 6.0 is going to eliminate half of the modules.  It's kinda unmanageable.

File: db/dcrpg/go.mod [hunk 5]
```diff
@@ -25,7 +25,7 @@ require (
 	github.com/decred/dcrdata/blockdata/v5 v5.0.1
 	github.com/decred/dcrdata/db/cache/v3 v3.0.1
 	github.com/decred/dcrdata/db/dbtypes/v2 v2.2.1
-	github.com/decred/dcrdata/explorer/types/v2 v2.1.1
+	github.com/decred/dcrdata/explorer/types/v2 v2.1.2
```
Comment (chappjc) [inline db/dcrpg/go.mod]: There is no v2.1.2 yet, just 2.1.1.  The replace will use the current version though.
git tag -l | grep explorer/types

Comment (peterzen) [reply db/dcrpg/go.mod]: Done, with the replaces in place it should be all good.

File: mempool/collector.go (hunk 3 above)
Comment (chappjc) [reply mempool/collector.go]: Nah, but I think stakebase/votes might have the same issue as they're generating coins too.

Comment (peterzen) [reply mempool/collector.go]: FeeRate is displayed for SSTX and regular TXs but not for votes; weird `feeRate` values coming from coinbase/stakebase txs won't be shown.  If you think it's warranted I can check `feeRate` coming back from `txhelpers.TxFeeRate` and maybe blank it out - what do you think?

Comment (chappjc) [reply mempool/collector.go]: Yeah, true.  It's good for this.

**Pull request merged**